crates/experimental-api/src/rpc.rs

```diff
@@ -103,6 +110,40 @@ impl IntoResponse for RpcError {
 struct Context {
     client: Arc<gen::client::Client>,
     url: String,
+    node: Arc<RwLock<NodeData>>,
+}
+
+impl Context {
+    pub async fn get_proper_block_id(
+        &self,
+        block_id: BlockId,
+    ) -> std::result::Result<BlockId, jsonrpc::Error> {
+        let l1_block_num = self.node.read().await.l1_block_number as i64;
+        match block_id {
+            gen::BlockId::BlockNumber { block_number } => {
+                let mut block_num = l1_block_num;
+                let user_request_block_num = *block_number.as_ref();
+                if user_request_block_num < block_num {
+                    block_num = user_request_block_num;
+                }
+                Ok(BlockId::BlockNumber {
+                    block_number: BlockNumber::try_new(block_num)?,
+                })
+            }
+            gen::BlockId::BlockHash { block_hash } => {
```

Comment:
For block hashes it might be trickier that that. Ideally we should resolve the block hash into the block number and check if that block number is acceptable.

Please create a follow up issue for that and put a comment here referencing the issue, just to avoid scope extension for this pull request.

Reply:
Created https://github.com/eigerco/beerus/issues/675 and added to backlog of the [beerus](https://github.com/orgs/eigerco/projects/27) github project